Refactor CloseAccountPage to functional component

bondydaa opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This refactors the CloseAccountPage into a functional component.

A few of the things in the existing component aren't necessary (ex: the lifecycle method) so I did not port those over. 

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/16298
PROPOSAL: GH_LINK_ISSUE(COMMENT)


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

1. Logged in and closed an account and ensured it closed properly
2. Logged in and went to the close account page and added a message and submitted the form and ensured the message showed up in the logs:
    ```
    2023-05-22T21:18:48.854700+00:00 expensidev2004 php-fpm: gNqtRg /api.php wickford@dog.com !ecash1.3.16-7! ?api? [hmmm] Account closed with parting message ~~ email: 'wickford@dog.com' message: 'fffffasdf'
    ```

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

1. Logged in, forced offline mode, went to the close account page and ensured I wasn't able to submit since this functionality requires network access.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
4. Upload an image via copy paste
5. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

1. Log into an account you are able to close without major repercussions (ie: not an admin/owner of workspace)
2. Force offline mode and attempt to close the account, ensure you're not allowed to submit the form.
3. Go back online, attempt to close the account and ensure you're able to submit it properly
4. When you are redirected to the login/sign in page ensure the Green Dot Road message of `Account closed successfully` appears correctly: <img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/0760845d-65c9-42d7-8254-5f4cefde7efa">
 

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/1d1e2bd6-d808-49f6-a06f-9d7adf3f38d2">
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/022c67fc-f1d6-43cf-bd4f-3019b58bb3e9">
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/aaf22d63-7173-4ae2-82b0-ca32e0e7ccd2">


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/82395fed-807e-4253-aa0b-a9a8a1f7bf20">
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/8e842f2f-9c01-4539-8afc-24e8269c469a">
<img width="1510" alt="image" src="https://github.com/Expensify/App/assets/4073354/dffdc2f1-68af-4204-a7f3-d7bfc4031ccb">
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/4c9330c0-3f86-44aa-8da4-e5a9a1afa576">


</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

bondydaa committed:
refactor to functional component

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-refactor-close-account

bondydaa committed:
remove unused method

bondydaa committed:
dont use useCallback here as its most likely not needed

bondydaa committed:
clean up imports

bondydaa commented:
okay publishing this for review.

One thing that was curious to me locally which I would love for someone to confirm is whether or not you get the proper "account was successfully closed" message on the log in page after closing the account. 

I could see in the onyx data the the data was present but I think pusher was failing for me locally and so the GBR wouldn't show up until I refreshed the page.

parasharrajat reviewed (changes requested):
Please fix the Lint issues.

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-refactor-close-account

bondydaa committed:
export default and clean up usage

bondydaa commented on src/libs/actions/CloseAccount.js:
```diff
@@ -2,13 +2,6 @@ import Onyx from 'react-native-onyx';
-/**
- * Clear CloseAccount error message to hide modal
- */
-function clearError() {
```
hmm you might be right, I noticed at one point in testing but then it went away after I'd reloaded and then I couldn't reproduce. Let me try again.

bondydaa replied:
Though don't think this method would actually stop the loader from showing since I think that is controlled by the `isLoading` key that's defined here: https://github.com/Expensify/App/blob/63f314c054d82ad0ee5e59ec5349d9b35d656bea/src/CONST.js#L730

bondydaa replied:
The more I noodle on this I think maybe we shouldn't be trying to address this here?

Really these props are all reacted-to by the `Form` component, this component is sort of just attempting to pass that data down along to it.

The only way to really solve it is to use the `useEffect` / lifecycle methods that were being used but those don't really make sense to add here since our component doesn't actually depend on them.

If we were to re-implement the lifecycle methods here we'd need `useEffect` which is just going to set some onyx data to the default because that's what the `<Form>` component will react to.

bondydaa committed:
fix default error key as defined by Onyx::handleException in web-e

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-refactor-close-account

bondydaa committed:
Revert "remove unused method"

This reverts commit bd80ad9d531d48a0aa29720cc3280cac8fe839b3.

bondydaa committed:
Revert "export default and clean up usage"

This reverts commit 270b71ce0b21fd6f38f635b98caac1a354023014.

bondydaa committed:
Revert "clean up imports"

This reverts commit df15ca036740d1efcb5243d43558d1ec1724609e.

bondydaa committed:
copy functionality based on lifecycle methods to not change behavior too much

bondydaa commented:
updated and reverted some of the broader changes and made this hopefully work more inline with how the class component was working. 

testing I still don't see the GBR "account closed successfully" but still 🙏 that's due to some weird pusher thing locally.

I did figure out one way you could trigger an error is to submit an IOU from the account your trying to close and then pay it from the account you submitted to 
<img width="1510" alt="image" src="https://github.com/Expensify/App/assets/4073354/76f7a053-ce97-4e0a-ad12-08d4f0263caa">

bondydaa commented on src/pages/settings/Security/CloseAccountPage.js:
```diff
@@ -37,105 +40,94 @@ const defaultProps = {
-            isConfirmModalVisible: false,
-            reasonForLeaving: '',
+    useEffect(() => {
+        CloseAccount.setDefaultData();
```
It's similar to what we were doing previously (though I don't love that we were doing it previously) but mostly the react docs suggest you should typically do something in the first part and undo it in the 2nd part so figure this was more inline with the "react way" but I don't have a super strong opinion, I think either is fine.

bondydaa replied:
oh after looking again... it jogged my memory a little bit.

I did it this way since that's how it was working before but ideally what we'd do is hook this up to the component directly in the `withOnyx` and pass these as actual props.

Since we were trying to change a little as possible I figured better to follow what was happening then we can clean everything else up after we're done with the refactor. 

Happy to change it still.

bondydaa commented:
ah I bet this wasn't throwing before because the default key was `error: ''` not `errors: ''`, i'll update that too.

bondydaa committed:
fix default type to stop console error

bondydaa committed:
remove setting default data on component mount, its not 100% necessary

bondydaa commented:
oh woohoo whatever was bork'd with pusher I think is fixed now too, got the GBR message after I closed the account locally 
<img width="1199" alt="image" src="https://github.com/Expensify/App/assets/4073354/2fa10818-de92-4cc1-9856-872759411e31">


updated with most of the latest suggestions but didn't add the empty dependencies array to the end of the `useEffect` hook since my understanding is that it's unnecessary but if we decided to always add it for some reason let me know and I can add that too.

bondydaa committed:
fix linter

bondydaa commented on src/pages/settings/Security/CloseAccountPage.js:
```diff
@@ -37,105 +40,89 @@ const defaultProps = {
-            reasonForLeaving: '',
-        };
-    }
+    useEffect(() => () => CloseAccount.clearError());
```
Hmm the linter failed and wanted me to fix this but I think it's kind of confusing looking now, ultimately we want to remove it anyways so I'll leave it up to the reviewers on whether or not we should leave it as is.

bondydaa replied:
you can see what it was like before here https://github.com/Expensify/App/pull/19328/commits/3830a3ed00437e48d8b90aced4d9261200e55e10

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-refactor-close-account

bondydaa committed:
fix styles with prettier

bondydaa commented on src/pages/settings/Security/CloseAccountPage.js:
```diff
@@ -37,105 +40,94 @@ const defaultProps = {
+            CloseAccount.clearError();
         };
-    }
+    });
```
@parasharrajat can you expand on why the `[]` is needed/necessary?

bondydaa replied:
doh nevermind GH UI didn't update with your last comment, reading now 🤦

bondydaa replied:
okay cool makes sense, thanks for that! updating now

bondydaa committed:
prevent execution of hook on every re-render by passing empty depedency array

bondydaa commented:
okay updated again, ready for review 🙏

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-refactor-close-account

chiragsalian reviewed (dismissed):
Tested on web and it seems to work good for me.
Overall the code also lgtm.

bondydaa commented:
😭 i'll fix the merge conflict tomorrow

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

cc: @chiragsalian 

:ribbon: :eyes: :ribbon:  C+ reviewed

chiragsalian commented:
oops sorry bondy looks like there is another conflict. Can you address and ping me on newDot so i can quickly review/merge it.

chiragsalian commented:
I could but then i won't be able to merge the PR since i pushed commits. Then we would have to find another internal engineer to review/merge it (or maybe you could merge it).

Either way, i hit up bondy about this last week and i believe he said the merge conflicts changes are a bit tricky so he would need some time for it. But with that said i would have expected it to be done by now. I see him online on slack everyday and according to his calendar he's working 50% this whole week except thursday. So @bondydaa, could you give us an ETA for this PR?

Let's wait till tomorrow to hear back from him. If we don't we'll plan something else.

chiragsalian commented:
I spoke to @bondydaa about this PR. He says his plate is pretty full and he won't be able to get to this PR this week. He says that there isn't much rush on this PR so he should be able to get to it next week.

bondydaa committed:
initial fix of merge conflicts, doesnt include all changes yet though

bondydaa committed:
apply changes from commit 38e49c372

bondydaa committed:
pull in change from commit b2dda336

bondydaa committed:
pull in changes from commit 1b36c75b17

bondydaa committed:
pull in changes from commit 3b5e279fe7

bondydaa commented:
okay believe i've fixed the merge conflict and pulled in changes from the various PRs. i tried to label each commit I pulled in manually as a separate commit to hopefully make it more clear and easier to follow instead of a single big "fix merge conflicts".

tested it and seems to still be working as expected so ready for another review now.

bondydaa committed:
pull in changes from commit 1e456f6788d1d

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5416206332) to staging by https://github.com/chiragsalian in version: 1.3.35-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

Julesssss commented:
Yeah, none of them are holding the release. Moving forward 👍

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5445522385) to production by https://github.com/Julesssss in version: 1.3.35-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5445522385) to production by https://github.com/Julesssss in version: 1.3.35-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

sakluger commented:
Hey folks, we have a new report (https://github.com/Expensify/App/issues/21951) that a user can trigger the form in Offline Mode with the Enter key on their keyboard. Is that a regression from this issue?

situchan commented on src/pages/settings/Security/CloseAccountPage.js:
```diff
@@ -38,103 +38,90 @@ const defaultProps = {
-                            confirmText={this.props.translate('common.yesContinue')}
-                            cancelText={this.props.translate('common.cancel')}
-                            shouldShowCancelButton
-                            danger
```
This prop is not moved correctly
